src/catalog/src/durable/persist.rs

```diff
@@ -1715,55 +1764,44 @@ impl UnopenedPersistCatalogState {
         T::Key: PartialEq + Eq + Debug + Clone,
         T::Value: Debug + Clone,
     {
-        let (_, prev) = retry(self, move |s| {
+        // We must fence out all other catalogs since we are writing.
+        let fence_updates = self.increment_epoch()?;
+        let prev_value = loop {
             let key = key.clone();
             let value = value.clone();
-            async {
-                let prev = s.debug_edit_inner::<T>(key, value).await;
-                (s, prev)
-            }
-        })
-        .await;
-        prev
-    }
```

Comment:
FWIW, this old retry logic is wrong and would have resulted in invalid retractions in the catalog. `s.debug_edit_inner()` increments the epoch field when calling `self.increment_epoch()` even if the compare and append fails. That causes us to think that the durably stored epoch is higher than it actually is resulting in an invalid retraction when we commit our fence updates.

I discovered this with some of the new tests I wrote.